File: packages/in-app-browser/ios/Plugin/Classes/Events/BrowserPageNavigationCompletedEvent.swift
import Foundation
import Capacitor

@objc public class BrowserPageNavigationCompletedEvent: NSObject, Result {
    let url: String

    init(url: String) {
        self.url = url
    }

    @objc public func toJSObject() -> AnyObject {
        var result = JSObject()
        result["url"] = url
        return result as AnyObject
    }
}

File: packages/in-app-browser/android/src/main/java/io/capawesome/capacitorjs/plugins/inappbrowser/classes/events/BrowserPageNavigationCompletedEvent.java
package io.capawesome.capacitorjs.plugins.inappbrowser.classes.events;

import androidx.annotation.NonNull;
import com.getcapacitor.JSObject;
import io.capawesome.capacitorjs.plugins.inappbrowser.interfaces.Result;

public class BrowserPageNavigationCompletedEvent implements Result {

    @NonNull
    private final String url;

    public BrowserPageNavigationCompletedEvent(@NonNull String url) {
        this.url = url;
    }

    @Override
    @NonNull
    public JSObject toJSObject() {
        JSObject result = new JSObject();
        result.put("url", url);
        return result;
    }
}
